tests/publiccloud/check_registercloudguest.pm

```diff
@@ -93,18 +93,18 @@ sub run {
     if (is_byos()) {
         $instance->run_ssh_command(cmd => 'sudo SUSEConnect --version');
         $instance->run_ssh_command(cmd => "sudo SUSEConnect $regcode_param");
-        $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest --clean");
+        $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest --clean") if (get_var('PUBLIC_CLOUD_QAM'));
     }
 
-    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param");
+    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param") if (get_var('PUBLIC_CLOUD_QAM'));
     if ($instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 600) == 0) {
         die('The list of zypper repositories is empty.');
     }
     if ($instance->run_ssh_command(cmd => 'sudo ls /etc/zypp/credentials.d/* | wc -l') == 0) {
         die('Directory /etc/zypp/credentials.d/ is empty.');
     }
 
-    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param --force-new");
+    $instance->run_ssh_command(cmd => "sudo ${path}registercloudguest $regcode_param --force-new") if (get_var('PUBLIC_CLOUD_QAM'));
     if ($instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', timeout => 600) == 0) {
```

Comment:
you can exclude this too because without line 107 you basically repeating what already was done in lines 100 and 103

Reply:
Can you re-check? I'm a bit confused now...